Library/Homebrew/cmd/log.rb

```diff
@@ -14,14 +14,24 @@ def log
 
   private
 
-  def git_log(path=nil)
+  def git_log(path = nil)
     if File.exist? "#{`git rev-parse --show-toplevel`.chomp}/.git/shallow"
       opoo <<-EOS.undent
         The git repository is a shallow clone therefore the filtering may be incorrect.
         Use `git fetch --unshallow` to get the full repository.
       EOS
     end
+
+    if ARGV.size > 2 then raise <<-EOS.undent
+      Usage is:
+        brew log formula
+      or:
+        brew log formula version
+    EOS
+    end
+
     args = ARGV.options_only
+    args += ["--grep", ARGV[1]] unless ARGV[1].nil?
```

Comment:
This is also referring to all arguments, not just the named arguments, so in legitimate examples like `brew log --oneline ffmpeg 2.8.1` this will actually grep for `ffmpeg`. My feeling is that it would be better to let the `log` method further up handle the user-supplied arguments and then pass them on to this `git_log` method. Here's a sketch of my idea:

```ruby
# In `log` further up:
  path = Formulary.path(ARGV.named.first)
  text = ARGV.named[1] if ARGV.named.size >= 2
  cd path.dirname # supports taps
  git_log path, text

def git_log(path = nil, text = nil)
  # Use `text` (if not `nil`) later on to construct `--grep` argument for `git`.
```